AllJoyn/Platform/Bridge/BridgeNET/BridgeNET.cpp

```diff
@@ -67,22 +67,71 @@ namespace BridgeNet
         std::shared_ptr<IBridge> bridge = IBridge::Get();
         bridge->AddAdapter(scriptAdapter);
         bridge->Initialize();
-    
+
         return S_OK;
     }
 
-    void BridgeJs::AddDevice(DeviceInfo^ info, 
-        String^ baseTypeXml, 
-        String^ script, 
+    // Wraps a shared_ptr to an IAdapterIORequest; preserves shared_ptr lifetime management
+    // of the request object and allows waiting for the request to complete.
+    ref class AdapterRequestAwaiter
+    {
+    public:
+        AdapterRequestAwaiter(const std::shared_ptr<IAdapterAsyncRequest>& request)
```

Comment:
Yeah I'm aware of the difference. It actually does get copied on the line below so the reference is OK in this case... Except I don't know what I was thinking using a pointer to a shared_ptr as the _requestPtr member, that should not be necessary, I'll clean that up.